src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -313,6 +320,29 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
       .sort()
       .every((eId, index) => eId === aIds[index]);
   }
+
+  private updateHealth(cy: any) {
```

Comment:
I didn't expect the health check to be performed at the request of the client.  I thought it would be done on the server-side.  I'm not really sure how I feel about being done this way, I guess it's OK but the HTTP traffic may get intense.  And I'm not sure how this plays with auto-refresh (or manual refresh for that matter).   I guess it can be applied incrementally to the graph.  Something makes me think this may be better on the back-end, we may want to consider it going forward but at least for now I think this is fine.

Reply:
well this was my biggest concern in terms of architecture as well and the main advantage of doing this way, as I explain in the PR description, is that this way it won't slow down the graph display, since all health information is gathered independently from graph generation. So if fetching health is slow, it will only impact health display.

Both approaches has pros and cons (as you say, higher http traffic), but I think the more we're adding data to fetch, the more it becomes advantageous to fetch extra-data asynchronously.

By the way I created an optimization subtask ( https://issues.jboss.org/browse/KIALI-911 ) that could be used, for instance, to fetch all service's health at once.